coral-common/src/main/java/com/linkedin/coral/common/TypeConverter.java

```diff
@@ -164,18 +161,8 @@ public static TypeInfo convert(RelDataType rType) {
   }
 
   public static TypeInfo convertStructType(RelDataType rType) {
-    List<TypeInfo> fTypes = Lists.transform(rType.getFieldList(), new Function<RelDataTypeField, TypeInfo>() {
-      @Override
-      public TypeInfo apply(RelDataTypeField f) {
-        return convert(f.getType());
-      }
-    });
-    List<String> fNames = Lists.transform(rType.getFieldList(), new Function<RelDataTypeField, String>() {
-      @Override
-      public String apply(RelDataTypeField f) {
-        return f.getName();
-      }
-    });
+    List<TypeInfo> fTypes = rType.getFieldList().stream().map(f -> convert(f.getType())).collect(Collectors.toList());
+    List<String> fNames = rType.getFieldList().stream().map(RelDataTypeField::getName).collect(Collectors.toList());
```

Comment:
I think there's a method in `RelDataType` to directly return field names if it is a struct, so we can avoid converting to stream again.

Reply:
Great catch! Thanks!